hack/run-tests.sh

```diff
@@ -13,452 +13,37 @@
-  printf "\n--performance\n\tFilter to run only the tests tagged with '@performance'. If not provided and the tag itself is not specified, these tests will be ignored."
-  printf "\n--load_factor {INT_VALUE}\n\tSet the tests load factor. Useful for the tests to take into account that the cluster can be overloaded, for example for the calculation of timeouts. Default value is 1."
-  printf "\n--local\n\tSpecify whether you run test in local using either a local or remote cluster."
-  printf "\n--ci {CI_NAME}\n\tSpecify whether you run test with ci, give also the name of the CI."
-  printf "\n--cr_deployment_only\n\tUse this option if you have no CLI to test against. It will use only direct CR deployments."
-  printf "\n--load_default_config\n\tTo be used if you want to directly use the default test config contained into ${SCRIPT_DIR}/../test/.default_config"
-  printf "\n--container_engine\n\tTo be used if you want to specify engine to interact with images and containers. Default is docker."
-  printf "\n--domain_suffix\n\tTo be used if you want to set a domain suffix for exposed services. Ignored when running tests on Openshift."
-  printf "\n--image_cache_mode\n\tUse this option to specify whether you want to use image cache for runtime images. Available options are 'always', 'never' or 'if-available'(default)."
-  printf "\n--http_retry_nb {INT_VALUE}\n\tSet the retry number for all HTTP calls in case it fails (and response code != 500). Default value is 3."
-  printf "\n--olm_namespace \n\tSet the namespace which is used for cluster scope operators. Default is 'openshift-operators'."
+echo '---- Retrieving kogito-operator hash from `go.mod` file ----'
+kogito_operator_hash=$(cat ../go.mod | grep 'github.com/kiegroup/kogito-operator' | awk -F'-' '{print $4}')
+echo "Got kogito-operator hash ${kogito_operator_hash}"
 
-  # operator information
-  printf "\n--operator_image {NAME}\n\tOperator image name. Default is 'quay.io/kiegroup/kogito-operator' one."
-  printf "\n--operator_tag {TAG}\n\tOperator image tag. Default is operator version."
-  printf "\n--operator_namespaced {TAG}\n\tSet to true to deploy Kogito operator into namespace used for scenario execution, false for cluster wide deployment. Default is false."
-  printf "\n--operator_installation_source {TAG}\n\tDefines installation source for the Kogito operator. Options are 'olm' and 'yaml'. Default is yaml."
-  printf "\n--operator_catalog_image {TAG}\n\tDefines image containing operator catalog. Needs to be specified only when operator_installation_source is 'olm'."
+tmp_dir=$(mktemp -d)
 
-  # files/binaries
-  printf "\n--operator_yaml_uri {URI}\n\tUrl or Path to kogito-operator.yaml file."
-  printf "\n--cli_path {PATH}\n\tPath to built CLI to test. Default is local built one."
+echo '---- Retrieving kogito-operator testing file ----'
+cd ${tmp_dir}
+git clone https://github.com/kiegroup/kogito-operator.git &> /dev/null
```

Comment:
Adding here `--depth 1` will tremendously speed the cloning process. As we only need the last revision. Or we can put there 5, just to be on the safe side. But locally it downloads ~870 kB instead of 35 MB.

Reply:
This would be a problem if there were some changes (one or multiple commits) in `kogito-operator` and the `go.mod` file of `rhpam-kogito-operator` would not be configured to the last commit of `kogito-operator`.

Reply:
But aren't we supposed to fail fast in that case and notify a user that they use out of date `kogito-operator` reference? Without this check a user can run out of date version without knowing it. Just an idea...

Reply:
yeah but if the current version of `rhpam-kogito-operator` is not compatible with the latest version of `kogito-operator` due to latest changes happening on that latter repo, then you are stuck and cannot run the tests.
The change of version in `go.mod` file should not be handled in the run-tests script, but separately.

Reply:
And if it is not compatible, do we have any other check that will notify us? Otherwise we may end up running `rhpam-kogito-operator` against an old version of `kogito-operator` for some time without noticing it. If we have the PR check adjusted for that, I am fine with it.

Reply:
A specific script should check that on PRs (GHAction or Jenkins) or there was also some discussion about having a bot updating the hash version daily ... but not implemented yet ...

In fact, this is the responsibility of one script (not existing yet I do get that)
All other scripts in this repository which needs `kogito-operator` files or other should then be based on that defined `hash` version and not the latest version to avoid any compatibility problems.
So `run-tests` script itself should only be based on that defined hash.

Reply:
OK, makes sense :)